py/runai_model_streamer/runai_model_streamer/distributed_streamer/partition.py

```
@@ -147,30 +152,35 @@ def partition_by_files(
-    requests_with_indices = list(enumerate(file_stream_requests))
+    # 1. Sort the FileChunks objects from largest to smallest total size.
     sorted_requests = sorted(
-        requests_with_indices,
-        key=lambda item: sum(item[1].chunks),
+        file_stream_requests,
+        key=lambda request: request.total_size(),
         reverse=True
     )
 
     # 2. Distribute the sorted FileChunks objects into n partitions.
     partitions: List[List[Tuple[FileChunks, Dict[int, Tuple[int, int, int]]]]] = [[] for _ in range(n)]
```

Ruff (0.16.0)

[error] Undefined name `Tuple`

(F821)